examples/ui-demo/src/components/modals/Erc20/Erc20Modal.tsx

```diff
@@ -0,0 +1,381 @@
+          } else {
+            console.warn("Fee estimation did not return expected data.");
+            setNetworkFee(0);
+          }
+        } catch (error) {
+          setNetworkFee(0);
+        }
+      } else if (isOpen && accountAddress && !isLoadingEthPrice) {
+        setNetworkFee(0);
+        if (!ethPriceData?.price) {
+          console.warn("ETH price not available for fee estimation.");
+        }
+      }
+    };
+
+    fetchAndEstimateFee();
+  }, [
+    isOpen,
+    accountAddress,
+    estimateMintNftFee,
+    ethPriceData?.price,
+    isLoadingEthPrice,
+    balance,
+  ]);
+
+  useEffect(() => {
+    resetMintNft();
+    resetMint();
+  }, [accountMode, resetMintNft, resetMint]);
+
+  const buyNftButtonEnabled =
+    readyToBuyNft && !isMintingNft && !isLoadingNftClient;
+
+  // Determine BagIcon color
+  let bagIconColor = "#CBD5E1";
+  if (mintNftTxHash) {
+    bagIconColor = "#475569";
+  } else if (buyNftButtonEnabled) {
+    bagIconColor = "white";
+  }
+
+  return (
+    <Dialog isOpen={isOpen} onClose={handleClose}>
+      <div className="akui-modal md:w-[607px] rounded-lg overflow-hidden">
+        <DynamicHeight>
+          <div className="p-5 flex flex-col gap-6">
+            {/* Header */}
+            <div className="flex justify-between items-start">
+              <div className="flex flex-col gap-1.5">
+                <h2 className="font-semibold text-xl leading-tight tracking-tighter text-fg-primary">
+                  Pay gas with any token
+                </h2>
+                <p className="text-sm font-normal leading-relaxed text-fg-secondary">
+                  Checkout with one click and a single token. USDC is an
+                  example.
+                </p>
+              </div>
+              <button
+                className="text-fg-secondary w-[40px] h-[40px] flex items-center justify-center hover:bg-btn-secondary rounded-md"
+                onClick={handleClose}
+              >
+                <XIcon className="w-[24px] h-[24px] lg:w-[16px] lg:h-[16px] stroke-fg-primary" />
+              </button>
+            </div>
+
+            {/* Body */}
+            <div className="flex flex-col md:flex-row gap-6">
+              {/* Image */}
+              <div className="md:w-1/2 rounded-lg overflow-hidden">
+                <div className="relative aspect-square">
+                  <Image
+                    src="/images/erc-20-sponsorship.jpg"
+                    alt="NFT"
+                    fill
+                    className="object-cover rounded-lg"
+                  />
+                </div>
+              </div>
+
+              {/* Details */}
+              {!isMintingNft && !mintNftTxHash && (
+                <div className="md:w-1/2 flex flex-col">
+                  {/* 1) Warning */}
+                  <div className="flex flex-col gap-4 pb-2">
+                    <div className="h-12">
+                      <Transition
```

Comment:
Added Transtion from headless ui to help with the animation.  Let me know if we don't want to use this but it definitely cleaned up some of the code